src/action.c

```diff
@@ -293,6 +306,48 @@ actions_contain_toggle_keybinds(struct wl_list *action_list)
 	return false;
 }
 
+/* Checks for *required* arguments */
+bool
+action_is_valid(struct action *action)
+{
+	char *arg_name = NULL;
+	switch (action->type) {
```

Comment:
We could just create a separate `get_arg_name()` function here like with:

```
case ACTION_TYPE_EXECUTE:
    return "command";
...
```

Reply:
The issue with a generalized function like that is that there may be multiple arguments per action.
In this specific case we only care for the required arguments which is at most one per action right now but that may change in the future. And as its only required in this new validation function I didn't think its worth adding another function for it.